### PR TITLE
Add graphql explorer to the generator

### PR DESCRIPTION
**Pull request by xt0rted**
I was playing around with this page earlier and found a number of unstyled parts due to a style sheet that wasn't being included in the generator.

I added the url in and started adding some of the missing colors but wanted to PR the changes to see if this was adding too many new selectors/rules in vs. manually styling it. There's some manual styles that can be removed but I didn't get to those yet.

This is still a work in progress but I haven't seen any issues elsewhere on the site from it.

### CODE REVIEW

File: tools/generate.js [hunk 1]
```diff
@@ -158,6 +164,7 @@ let mappings = {
 
   "color: #0366d6": "color: /*[[base-color]]*/ #4f8cc9", // needs to be after #333
   "color: #1074e7": "color: /*[[base-color]]*/ #4f8cc9",
+  "color: #1F61A0": "color: /*[[base-color]]*/ #4f8cc9", //  graphql explorer
```
Comment (xt0rted) [inline tools/generate.js]: @silverwind are these values case sensitive? A few of the colors in this style sheet use all caps which is why they're using it here too.

Comment (the-j0k3r) [reply tools/generate.js]: I dont change the hex case from the default stylesheets when adding them to generator, though I always use lower case for replacement colors as its easier to read 8 vs B etc.

**Comment by xt0rted** [comment]
Looks like I found the first big issue with these changes. The style sheet used on this domain has classes with colors that don't match github.com now. Very specifically the `.link-gray` class uses `#767676` while on github.com it's `#586069`. This causes two auto-generated rules to be added and the one from the graphql explorer takes priority which darkens the text in a number of spots.

image

image

**Comment by the-j0k3r** [comment]
> The style sheet used on this domain has classes with colors that don't match github.com now. Very specifically the .link-gray class uses #767676 while on github.com it's #586069. This causes two auto-generated rules to be added and the one from the graphql explorer takes priority which darkens the text in a number of spots.

What do you think about these sorts of additions going to their own @moz-rule so as to avoid these conflicts? I did the same with another section a while back for same reasons though I used the autogenerated rules to build the specific section.

Thats until the generator can handle those sections automatically.

**Comment by xt0rted** [comment]
It looks like that's what'll have to happen here too. Once everything is in place I can copy all the rules into their own section, then remove the colors from the generator for now.

File: tools/generate.js (hunk 1 above)
Comment (silverwind) [reply tools/generate.js]: They are compared insensitively, thought I do prefer lowercase in this file.

File: tools/generate.js [hunk 2]
```diff
@@ -283,6 +293,7 @@ const sources = [
   {url: "https://github.com"},
   {url: "https://gist.github.com"},
   {url: "https://help.github.com"},
+  {url: "https://graphql-explorer.githubapp.com/"},
```
Comment (silverwind) [inline tools/generate.js]:
```suggestion
  {
    url: "https://graphql-explorer.githubapp.com/", 
    prefix: ".graphql-container",
    match: [".graphql-container"]
  },
```

**Comment by silverwind** [comment]
Should be scoped to a selector like above. html and body don't have any usable properties, so `.graphql-container` seems the best we can do.

File: tools/generate.js (hunk 2 above)
Comment (the-j0k3r) [reply tools/generate.js]: @xt0rted commit @silverwind suggestion and regenerate, if it works (it should just like mobile styles go ahead and merge.

**Comment by the-j0k3r** [comment]
@xt0rted rebase on master and regenerate I committed @silverwinds suggestion in https://github.com/StylishThemes/GitHub-Dark/pull/1018/commits/023830c632d291c3c008d5f3741c6390842d75a8

File: tools/generate.js [hunk 3]
```diff
@@ -283,6 +293,11 @@ const sources = [
   {url: "https://github.com"},
   {url: "https://gist.github.com"},
   {url: "https://help.github.com"},
+  {
+    url: "https://graphql-explorer.githubapp.com/",
```
Comment (the-j0k3r) [inline tools/generate.js]:
```suggestion
    url: "https://graphql-explorer.githubapp.com/",
```

**Comment by the-j0k3r** [comment]
Ive moved this out of draft so to get it moving.

**Comment by silverwind** [comment]
Looking at it again, we might be able to use `html[prefix*="ogp.me"]` for the prefix. That URL refers to some graph protocol, so should be unique enough I guess.

**Comment by silverwind** [comment]
Also, link to it because we don't have one yet in this thread: https://developer.github.com/v4/explorer

**Comment by silverwind** [comment]
Implemented this. Not really tested as the actual application looks broken to me (just shows `Loading...`).

**Pull request closed without merging**